lib/ansible/modules/identity/cyberark/cyberark_user.py

```diff
@@ -0,0 +1,461 @@
+              this parameter is required when state is present -- Add User.
+    new_password:
+        description:
+            - The user updated password. Make sure that this password meets the password policy requirements.
+    email:
+        description:
+            - The user email address.
+    first_name:
+        description:
+            - The user first name.
+    last_name:
+        description:
+            - The user last name.
+    change_password_on_the_next_logon:
+        type: bool
+        default: 'no'
+        description:
+            - Whether or not the user must change their password in their next logon.
+              Valid values = true/false.
+    expiry_date:
+        description:
+            - The date and time when the user account will expire and become disabled.
+    user_type_name:
+        default: EPVUser
+        description:
+            - The type of user.
+    disabled:
+        type: bool
+        default: 'no'
+        description:
+            - Whether or not the user will be disabled. Valid values = true/false.
+    location:
+        description:
+            - The Vault Location for the user.
+    group_name:
+        description:
+            - The name of the group the user will be added to.
+'''
+
+EXAMPLES = '''
+- name: Logon to CyberArk Vault using PAS Web Services SDK
+  cyberark_authentication:
+    api_base_url: "https://components.cyberark.local"
+    use_shared_logon_authentication: true
+
+- name: Create user & immediately add it to a group
+  cyberark_user:
+    username: "username"
+    initial_password: "password"
+    user_type_name: "EPVUser"
+    change_password_on_the_next_logon: false
+    group_name: "GroupOfUsers"
+    state: present
+    cyberark_session: "{{ cyberark_session }}"
+
+- name: Make sure user is present and reset user credential if present
+  cyberark_user:
+    username: "Username"
+    new_password: "password"
+    disabled: false
+    state: present
+    cyberark_session: "{{ cyberark_session }}"
+
+- name: Logoff from CyberArk Vault
+  cyberark_authentication:
+    state: absent
+    cyberark_session: "{{ cyberark_session }}"
+'''
+
+RETURN = '''
+changed:
+    description: Whether there was a change done.
+    type: bool
+    returned: always
+cyberark_user:
+    description: Dictionary containing result properties.
+    returned: always
+    type: dict
+    sample:
+        result:
+            description: user properties when state is present
+            type: dict
+            returned: success
+status_code:
+    description: Result HTTP Status code
+    returned: success
+    type: int
+    sample: 200
+'''
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+from ansible.module_utils.six.moves import http_client as httplib
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.urls import open_url
+
+
+def userDetails(module):
```

Comment:
Just style again.  We use this style for most identifiers (including function names):
```
def user_details(module):
```

For classes we use CapWord like:
``` python
class PasswordSession:
    pass
```

We don't use initial-lower camelCase anywhere.